Tests/SwiftParserTest/translated/ErrorsTests.swift

```diff
@@ -238,68 +243,103 @@ final class ErrorsTests: XCTestCase {
       func dupThrows2(_ f: () throws -> 1️⃣rethrows Int) {}
       """,
       diagnostics: [
-        // TODO: Old parser expected error on line 1: 'rethrows' has already been specified, Fix-It replacements: 35 - 44 = ''
-        DiagnosticSpec(message: "expected type in function type"),
-        DiagnosticSpec(message: "unexpected code 'rethrows Int' in parameter clause"),
-      ]
+        DiagnosticSpec(message: "'rethrows' may only occur before '->'", fixIts: ["remove redundant 'rethrows'"])
+      ],
+      fixedSource: """
+        func dupThrows2(_ f: () throws -> Int) {}
+        """
     )
   }
 
-  func testErrors15() {
+  func testErrors15a() {
     AssertParse(
       """
-      func dupThrows3() {
-        _ = { () try throws in }
-        _ = { () throws -> Int 1️⃣throws in }
-      }
+      _ = { () 1️⃣try throws in }
       """,
       diagnostics: [
-        // TODO: Old parser expected error on line 2: expected throwing specifier; did you mean 'throws'?, Fix-It replacements: 12 - 15 = 'throws'
-        // TODO: Old parser expected error on line 2: 'throws' has already been specified, Fix-It replacements: 16 - 23 = ''
-        // TODO: Old parser expected error on line 3: 'throws' has already been specified, Fix-It replacements: 26 - 33 = ''
-        DiagnosticSpec(message: "expected 'in' in closure signature"),
-        DiagnosticSpec(message: "unexpected code 'throws in' in closure"),
-      ]
+        DiagnosticSpec(message: "'try' conflicts with 'throws'", fixIts: ["remove redundant 'try'"])
+      ],
+      fixedSource: """
+        _ = { () throws in }
+        """
+    )
+  }
+
+  func testErrors15b() {
+    AssertParse(
+      """
+      _ = { () throws -> Int 1️⃣throws in }
+      """,
+      diagnostics: [
+        DiagnosticSpec(message: "'throws' may only occur before '->'", fixIts: ["remove redundant 'throws'"])
+      ],
+      fixedSource: """
+        _ = { () throws -> Int in }
+        """
     )
   }
 
   func testErrors16() {
     AssertParse(
       """
       func incompleteThrowType() {
-        // FIXME: Bad recovery for incomplete function type.
         let _: () 1️⃣throws
       }
       """,
       diagnostics: [
-        // TODO: Old parser expected error on line 3: consecutive statements on a line must be separated by ';'
-        // TODO: Old parser expected error on line 3: expected expression
         DiagnosticSpec(message: "unexpected 'throws' keyword in function")
```

Comment:
I assume this is parsed as `let _: () // throws` ie. a variable with void type?

Reply:
Correct. Added a substructure test